### PR TITLE
Use core's new fifo functions

### PR DESCRIPTION
**Pull request by kneth**
Realm Core v5.23.2 adds a couple of new FIFO functions.

### CODE REVIEW

**Comment by nhachicha** [comment]
be careful, once merged [realm_coordinator.cpp](https://github.com/realm/realm-object-store/blob/core-6/src/impl/realm_coordinator.cpp#L44) in `core-6` branch needs to be updated as well to use the include from Core 
```Diff
--- a/src/impl/realm_coordinator.cpp
+++ b/src/impl/realm_coordinator.cpp
@@ -41,7 +41,7 @@
 #include <realm/db.hpp>
 #include <realm/history.hpp>
 #include <realm/string_data.hpp>
-#include <util/fifo.hpp>
+#include <realm/util/fifo_helper.hpp>
 
 #include <algorithm>
 #include <unordered_map>

```

**Comment by kneth** [comment]
@nhachicha I hope that #798 will take of that :smile:.

**Comment by cmelchior** [comment]
After restarting the test a few times, they still seem to fail with invalid top refs: `Unable to open a realm at path '/var/folders/01/

  qdkfxkd167338d33x5c6bt540000gl/T/realm-object-server/io.realm.object-server-

  utility/metadata/sync_metadata.realm': Invalid top array (ref: 31432, size:

  11) Path:.
`

File: workflow/test_coverage.sh [hunk 1]
```diff
@@ -13,6 +13,12 @@ fi
 
 set -e
 
+if [ "${TMPDIR}" = "" ]; then
+  TMPDIR="/tmp"
+fi
+
+echo "TMPDIR: ${TMPDIR}"
+rm -rf ${TMP_DIR}/realm*
```
Comment (kneth) [inline workflow/test_coverage.sh]:
```suggestion
rm -rf ${TMPDIR}/realm*
```

**Pull request merged**